tools/cdb2sql/cdb2sql.cpp

```diff
@@ -117,11 +117,12 @@ void dumpstring(FILE *f, char *s, int quotes, int quote_quotes)
         fprintf(f, "'");
 }
 
-#define verbose_print(...)                                                     \
-    {                                                                          \
+#define verbose_print(fmt, args...)                                            \
+    do {                                                                       \
         if (verbose)                                                           \
-            fprintf(stderr, __VA_ARGS__);                                      \
-    }
+            fprintf(stderr, "td 0x%u %s:%d " fmt, (uint32_t)pthread_self(),    \
```

Comment:
Unfortunately this one does not compile on my Mac (with a bb-homebrew formula). The error is as follows:

```
 27%] Building CXX object tools/cdb2sql/CMakeFiles/cdb2sql.dir/cdb2sql.cpp.o
cd /tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/build/tools/cdb2sql && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++  -DCOMDB2_ROOT=/usr/local/Cellar/comdb2/r7a -DCOMDB2_VERSION=\"2\" -DNEWSI_ASOF_USE_TEMPTABLE -DNEWSI_MEMPOOL -DPER_THREAD_MALLOC -DWITH_SSL=1 -D_DARWIN_C_SOURCE -D_FILE_OFFSET_BITS=64 -D_LINUX_SOURCE -D_XOPEN_SOURCE=600 -I/tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/util -I/tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/bbinc -I/tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/cdb2api -I/usr/local/opt/openssl/include -I/usr/local/opt/readline/include  -DNDEBUG   -std=gnu++11 -o CMakeFiles/cdb2sql.dir/cdb2sql.cpp.o -c /tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/tools/cdb2sql/cdb2sql.cpp

/tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/tools/cdb2sql/cdb2sql.cpp:1004:9: error: cast from pointer to smaller type 'uint32_t' (aka 'unsigned
 int') loses information
        verbose_print("calling cdb2_open\n");
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/tools/cdb2sql/cdb2sql.cpp:123:51: note: expanded from macro 'verbose_print'
            fprintf(stderr, "td 0x%u %s:%d " fmt, (uint32_t)pthread_self(),    \
                                                  ^~~~~~~~~~~~~~~~~~~~~~~~
/tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/tools/cdb2sql/cdb2sql.cpp:1087:13: error: cast from pointer to smaller type 'uint32_t' (aka 'unsigne
d int') loses information
            verbose_print("setting bind parameter\n");
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/tools/cdb2sql/cdb2sql.cpp:123:51: note: expanded from macro 'verbose_print'
            fprintf(stderr, "td 0x%u %s:%d " fmt, (uint32_t)pthread_self(),    \
                                                  ^~~~~~~~~~~~~~~~~~~~~~~~
/tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/tools/cdb2sql/cdb2sql.cpp:1113:13: error: cast from pointer to smaller type 'uint32_t' (aka 'unsigne
d int') loses information
            verbose_print(
            ^~~~~~~~~~~~~~
/tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/tools/cdb2sql/cdb2sql.cpp:123:51: note: expanded from macro 'verbose_print'
            fprintf(stderr, "td 0x%u %s:%d " fmt, (uint32_t)pthread_self(),    \
                                                  ^~~~~~~~~~~~~~~~~~~~~~~~
/tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/tools/cdb2sql/cdb2sql.cpp:1236:5: error: cast from pointer to smaller type 'uint32_t' (aka 'unsigned int') loses information
    verbose_print("processing line sql '%.30s...'\n", sql);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/tools/cdb2sql/cdb2sql.cpp:123:51: note: expanded from macro 'verbose_print'
            fprintf(stderr, "td 0x%u %s:%d " fmt, (uint32_t)pthread_self(),    \
                                                  ^~~~~~~~~~~~~~~~~~~~~~~~
/tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/tools/cdb2sql/cdb2sql.cpp:1302:5: error: cast from pointer to smaller type 'uint32_t' (aka 'unsigned int') loses information
    verbose_print("processing typed statement arguments\n");
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/tools/cdb2sql/cdb2sql.cpp:123:51: note: expanded from macro 'verbose_print'
            fprintf(stderr, "td 0x%u %s:%d " fmt, (uint32_t)pthread_self(),    \
                                                  ^~~~~~~~~~~~~~~~~~~~~~~~
/tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/tools/cdb2sql/cdb2sql.cpp:1606:9: error: cast from pointer to smaller type 'uint32_t' (aka 'unsigned int') loses information
        verbose_print("process_line error=%d\n", error);
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/comdb2-20180919-95514-e0d6e1/comdb2-master/tools/cdb2sql/cdb2sql.cpp:123:51: note: expanded from macro 'verbose_print'
            fprintf(stderr, "td 0x%u %s:%d " fmt, (uint32_t)pthread_self(),    \
                                                  ^~~~~~~~~~~~~~~~~~~~~~~~
```